conans/test/unittests/client/tools/cppstd_required_test.py

```diff
@@ -0,0 +1,183 @@
+    def test_check_min_cppstd_from_settings_with_extension(self, cppstd):
+        """ current cppstd in settings must has GNU extension when extensions is enabled
+        """
+        with mock.patch("platform.system", mock.MagicMock(return_value="Linux")):
+            with mock.patch.object(OSInfo, '_get_linux_distro_info'):
+                self.conanfile.settings.values["compiler.cppstd"] = "gnu17"
+                check_min_cppstd(self.conanfile, cppstd, True)
+
+                self.conanfile.settings.values["compiler.cppstd"] = "17"
+                with self.assertRaises(ConanInvalidConfiguration) as raises:
+                    check_min_cppstd(self.conanfile, cppstd, True)
+                self.assertEqual("Current cppstd (17) does not have GNU extensions, which is "
+                                 "required on Linux platform.", str(raises.exception))
+
+    @parameterized.expand(["98", "gnu98", "11", "gnu11", "14", "gnu14", "17", "gnu17"])
+    def test_check_min_cppstd_from_settings_with_extension_windows(self, cppstd):
+        """ GNU extensions has no effect on Windows for check_min_cppstd
+        """
+        with mock.patch("platform.system", mock.MagicMock(return_value="Windows")):
+            self.conanfile.settings.values["compiler.cppstd"] = "gnu17"
+            check_min_cppstd(self.conanfile, cppstd, True)
+
+            self.conanfile.settings.values["compiler.cppstd"] = "17"
+            check_min_cppstd(self.conanfile, cppstd, True)
+
+    def test_check_min_cppstd_unsupported_standard(self):
+        """ check_min_cppstd must raise when the compiler does not support a standard
+        """
+        del self.conanfile.settings.values["compiler.cppstd"]
+        with self.assertRaises(ConanInvalidConfiguration) as raises:
+            check_min_cppstd(self.conanfile, "42", False)
+        self.assertEqual("Current compiler does not support the required c++ standard (42).",
+                         str(raises.exception))
+
+    def test_check_min_cppstd_gnu_compiler_extension(self):
+        """ Current compiler must support GNU extension on Linux when extensions is required
+        """
+        del self.conanfile.settings.values["compiler.cppstd"]
+        with mock.patch("platform.system", mock.MagicMock(return_value="Linux")):
+            with mock.patch.object(OSInfo, '_get_linux_distro_info'):
+                with mock.patch("conans.client.tools.settings.cppstd_flag", return_value="42"):
+                    with self.assertRaises(ConanInvalidConfiguration) as raises:
+                        check_min_cppstd(self.conanfile, "gnu42", True)
+                    self.assertEqual("Current compiler does not support GNU extensions.",
+                                     str(raises.exception))
+
+
+class ValidMinCppstdTests(unittest.TestCase):
+
+    def setUp(self):
+        self.conanfile = self._create_conanfile("gcc", "9", "Linux", "17", "libcxx")
```

Comment:
Move this to each test function (no need for the `conanfile` member)

Reply:
done